In [8]:
import pandas as pd

# Load the IPL data into a pandas dataframe

In [9]:
data = pd.read_csv('C:\\Users\\Bala Vignesh.A\\Desktop\\SCMA 632\\Cricket_data.csv')


In [10]:
print(data)

      season       id                                          name  \
0        NaN  1370350          Chennai Super Kings v Gujarat Titans   
1        NaN  1370351         Lucknow Super Giants v Mumbai Indians   
2        NaN  1370352                                     TBC v TBC   
3        NaN  1370353                                     TBC v TBC   
4     2023.0  1359544  Royal Challengers Bangalore v Gujarat Titans   
...      ...      ...                                           ...   
1027  2008.0   336012  Royal Challengers Bangalore v Mumbai Indians   
1028  2008.0   336019            Kings XI Punjab v Rajasthan Royals   
1029  2008.0   336038           Delhi Daredevils v Rajasthan Royals   
1030  2008.0   336039         Chennai Super Kings v Kings XI Punjab   
1031  2008.0   336040        Chennai Super Kings v Rajasthan Royals   

      short_name                                        description home_team  \
0       CSK v GT  Qualifier 1 (N), Indian Premier League at Chen..

In [17]:
print(ct_data)

Empty DataFrame
Columns: [season, id, name, short_name, description, home_team, away_team, toss_won, decision, 1st_inning_score, 2nd_inning_score, winner, result, start_date, end_date, venue_id, venue_name, home_captain, away_captain, pom, points, super_over, home_overs, home_runs, home_wickets, home_boundaries, away_overs, away_runs, away_wickets, away_boundaries, highlights, home_key_batsman, home_key_bowler, home_playx1, away_playx1, away_key_batsman, away_key_bowler, match_days, umpire1, umpire2, tv_umpire, referee, reserve_umpire]
Index: []

[0 rows x 43 columns]


In [11]:
(data['home_key_batsman'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: home_key_batsman, Length: 1032, dtype: bool

In [12]:
(data['home_key_bowler'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: home_key_bowler, Length: 1032, dtype: bool

In [13]:
(data['home_playx1'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: home_playx1, Length: 1032, dtype: bool

In [14]:
(data['away_playx1'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: away_playx1, Length: 1032, dtype: bool

In [15]:
(data['away_key_batsman'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: away_key_batsman, Length: 1032, dtype: bool

In [16]:
 (data['away_key_bowler'] == 'Rahul Tewatia')

0       False
1       False
2       False
3       False
4       False
        ...  
1027    False
1028    False
1029    False
1030    False
1031    False
Name: away_key_bowler, Length: 1032, dtype: bool

In [17]:
columns_to_check = ['home_key_batsman', 'home_key_bowler', 'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler']

filter_mask = data[columns_to_check].apply(lambda x: x.str.contains('Rahul Tewatia')).any(axis=1)

rahul_tewatia_data = data[filter_mask]

print(rahul_tewatia_data)

     season       id                                            name  \
4    2023.0  1359544    Royal Challengers Bangalore v Gujarat Titans   
12   2023.0  1359536            Gujarat Titans v Sunrisers Hyderabad   
17   2023.0  1359531                 Mumbai Indians v Gujarat Titans   
23   2023.0  1359525           Gujarat Titans v Lucknow Super Giants   
26   2023.0  1359522               Rajasthan Royals v Gujarat Titans   
..      ...      ...                                             ...   
525  2015.0   829741              Rajasthan Royals v Kings XI Punjab   
532  2015.0   829755        Kolkata Knight Riders v Rajasthan Royals   
592  2014.0   733979        Rajasthan Royals v Kolkata Knight Riders   
602  2014.0   733999  Royal Challengers Bangalore v Rajasthan Royals   
619  2014.0   734033              Kings XI Punjab v Rajasthan Royals   

    short_name                                        description home_team  \
4     RCB v GT  70th Match (N), Indian Premier League at

In [35]:
# Select only the rows that match the filter criteria
rahul_tewatia_alone_data = data[filter_mask]

# Count the number of rows where 'Rahul Tewatia' appears alone
num_alone = rahul_tewatia_alone_data.shape[0]

print(f"Number of rows where 'Rahul Tewatia' appears alone: {num_alone}")

Number of rows where 'Rahul Tewatia' appears alone: 79


In [18]:
columns_to_check = ['home_key_batsman', 'home_key_bowler', 'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler']

In [19]:
filter_mask = data[columns_to_check].apply(lambda x: x.str.count('Rahul Tewatia') == 1).any(axis=1)

In [20]:
print(data.columns)

Index(['season', 'id', 'name', 'short_name', 'description', 'home_team',
       'away_team', 'toss_won', 'decision', '1st_inning_score',
       '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date',
       'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom',
       'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets',
       'home_boundaries', 'away_overs', 'away_runs', 'away_wickets',
       'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler',
       'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler',
       'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee',
       'reserve_umpire'],
      dtype='object')


In [21]:
data.columns = data.columns.str.strip()

In [22]:
print(data.columns)

Index(['season', 'id', 'name', 'short_name', 'description', 'home_team',
       'away_team', 'toss_won', 'decision', '1st_inning_score',
       '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date',
       'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom',
       'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets',
       'home_boundaries', 'away_overs', 'away_runs', 'away_wickets',
       'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler',
       'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler',
       'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee',
       'reserve_umpire'],
      dtype='object')


# Assuming 'data' is the DataFrame containing the IPL data

### Create a new DataFrame with the required columns

In [23]:
ipl_data = data[['season', 'id', 'name', 'short_name', 'description', 'home_team', 'away_team', 'toss_won', 'decision', '1st_inning_score', '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date', 'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom', 'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets', 'home_boundaries', 'away_overs', 'away_runs', 'away_wickets', 'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler', 'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler', 'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee', 'reserve_umpire']]


In [51]:
print(ipl_data)

      season       id                                          name  \
0        NaN  1370350          Chennai Super Kings v Gujarat Titans   
1        NaN  1370351         Lucknow Super Giants v Mumbai Indians   
2        NaN  1370352                                     TBC v TBC   
3        NaN  1370353                                     TBC v TBC   
4     2023.0  1359544  Royal Challengers Bangalore v Gujarat Titans   
...      ...      ...                                           ...   
1027  2008.0   336012  Royal Challengers Bangalore v Mumbai Indians   
1028  2008.0   336019            Kings XI Punjab v Rajasthan Royals   
1029  2008.0   336038           Delhi Daredevils v Rajasthan Royals   
1030  2008.0   336039         Chennai Super Kings v Kings XI Punjab   
1031  2008.0   336040        Chennai Super Kings v Rajasthan Royals   

      short_name                                        description home_team  \
0       CSK v GT  Qualifier 1 (N), Indian Premier League at Chen..

# Extract the batsman, ball, runs, and wickets per player per match

In [24]:
ipl_data_batting = ipl_data[['season', 'id', 'name', 'short_name', 'description', 'home_team', 'away_team', 'toss_won', 'decision', '1st_inning_score', '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date', 'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom', 'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets', 'home_boundaries', 'home_key_batsman', 'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee', 'reserve_umpire']].rename(columns={'home_key_batsman': 'Player'})
ipl_data_bowling = ipl_data[['season', 'id', 'name', 'short_name', 'description', 'home_team', 'away_team', 'toss_won', 'decision', '1st_inning_score', '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date', 'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom', 'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets', 'home_boundaries', 'home_key_bowler', 'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee', 'reserve_umpire']].rename(columns={'home_key_bowler': 'Player'})
ipl_data_batting['Runs'] = ipl_data_batting['home_runs']
ipl_data_bowling['Wickets'] = ipl_data_bowling['home_wickets']
ipl_data_batting['Balls'] = ipl_data_batting['home_overs'] * 6
ipl_data_bowling['Balls'] = ipl_data_bowling['home_overs'] * 6

# Combine the batting and bowling data

In [26]:
ipl_data_player = pd.concat([ipl_data_batting, ipl_data_bowling], axis=0)


# Group by IPL round, player, and player type, and calculate total runs and wickets

In [27]:
ipl_data_grouped = ipl_data_player.groupby(['season', 'Player']).agg({'Runs': 'sum', 'Wickets': 'sum', 'Balls': 'sum'}).reset_index()

# Find the top three run-getters and top three wicket-takers in each IPL round

In [29]:
top_run_getters = ipl_data_grouped.groupby('season').apply(lambda x: x.nlargest(3, 'Runs')).reset_index(drop=True)
top_wicket_takers = ipl_data_grouped.groupby('season').apply(lambda x: x.nlargest(3, 'Wickets')).reset_index(drop=True)

# Print the results

In [30]:
print("Top Run-Getters in Each IPL Round:")
print(top_run_getters)
print("\nTop Wicket-Takers in Each IPL Round:")
print(top_wicket_takers)

Top Run-Getters in Each IPL Round:
    season                           Player   Runs  Wickets  Balls
0   2008.0    Gautam Gambhir,Shikhar Dhawan  692.0      0.0  450.6
1   2008.0     Shaun Marsh,Kumar Sangakkara  356.0      0.0  235.8
2   2008.0    Graeme Smith,Swapnil Asnodkar  307.0      0.0  205.2
3   2009.0          Matthew Hayden,MS Dhoni  482.0      0.0  342.6
4   2009.0        Yuvraj Singh,Irfan Pathan  291.0      0.0  240.0
5   2009.0      AB de Villiers,David Warner  243.0      0.0  213.6
6   2010.0     Ambati Rayudu,Saurabh Tiwary  390.0      0.0  240.0
7   2010.0        Murali Vijay,Albie Morkel  246.0      0.0  120.0
8   2010.0     Jacques Kallis,Robin Uthappa  204.0      0.0  111.0
9   2011.0      Michael Hussey,Murali Vijay  325.0      0.0  240.0
10  2011.0   Ambati Rayudu,Sachin Tendulkar  283.0      0.0  240.0
11  2011.0       Shaun Marsh,Dinesh Karthik  282.0      0.0  240.0
12  2012.0  Gautam Gambhir,Brendon McCullum  498.0      0.0  360.0
13  2012.0          Faf du 

In [32]:
import scipy.stats as stats
import matplotlib.pyplot as plt

# Extract the top three batsmen and bowlers in the last three IPL tournaments

In [33]:
print("Top Run-Getters in Each IPL Round:")


Top Run-Getters in Each IPL Round:


In [67]:
print(top_run_getters)

    season                           Player   Runs  Wickets  Balls
0   2008.0    Gautam Gambhir,Shikhar Dhawan  692.0      0.0  450.6
1   2008.0     Shaun Marsh,Kumar Sangakkara  356.0      0.0  235.8
2   2008.0    Graeme Smith,Swapnil Asnodkar  307.0      0.0  205.2
3   2009.0          Matthew Hayden,MS Dhoni  482.0      0.0  342.6
4   2009.0        Yuvraj Singh,Irfan Pathan  291.0      0.0  240.0
5   2009.0      AB de Villiers,David Warner  243.0      0.0  213.6
6   2010.0     Ambati Rayudu,Saurabh Tiwary  390.0      0.0  240.0
7   2010.0        Murali Vijay,Albie Morkel  246.0      0.0  120.0
8   2010.0     Jacques Kallis,Robin Uthappa  204.0      0.0  111.0
9   2011.0      Michael Hussey,Murali Vijay  325.0      0.0  240.0
10  2011.0   Ambati Rayudu,Sachin Tendulkar  283.0      0.0  240.0
11  2011.0       Shaun Marsh,Dinesh Karthik  282.0      0.0  240.0
12  2012.0  Gautam Gambhir,Brendon McCullum  498.0      0.0  360.0
13  2012.0          Faf du Plessis,MS Dhoni  368.0      0.0  2

In [34]:
print(top_run_getters.head(3))

   season                         Player   Runs  Wickets  Balls
0  2008.0  Gautam Gambhir,Shikhar Dhawan  692.0      0.0  450.6
1  2008.0   Shaun Marsh,Kumar Sangakkara  356.0      0.0  235.8
2  2008.0  Graeme Smith,Swapnil Asnodkar  307.0      0.0  205.2


In [35]:
print("\nTop Wicket-Takers in Each IPL Round:")


Top Wicket-Takers in Each IPL Round:


In [36]:
print(top_wicket_takers)

    season                                Player  Runs  Wickets  Balls
0   2008.0             Praveen Kumar,Zaheer Khan   0.0     18.0  235.2
1   2008.0          Farveez Maharoof,Amit Mishra   0.0     10.0   96.6
2   2008.0            Piyush Chawla,Irfan Pathan   0.0     10.0  120.0
3   2009.0            Anil Kumble,Jacques Kallis   0.0     17.0  236.4
4   2009.0      Lakshmipathy Balaji,Albie Morkel   0.0     14.0  240.0
5   2009.0                RP Singh,Venugopal Rao   0.0     13.0  240.0
6   2010.0           Zaheer Khan,Harbhajan Singh   0.0     17.0  349.8
7   2010.0          Doug Bollinger,Shadab Jakati   0.0     16.0  240.0
8   2010.0                 RP Singh,Pragyan Ojha   0.0     16.0  224.4
9   2011.0             Rahul Sharma,Yuvraj Singh   0.0     16.0  240.0
10  2011.0                Johan Botha,Amit Singh   0.0     13.0  224.4
11  2011.0            Munaf Patel,Lasith Malinga   0.0     11.0  235.2
12  2012.0                Dale Steyn,Amit Mishra   0.0     14.0  230.4
13  20

In [74]:
print(top_wicket_takers.head(3))

   season                        Player  Runs  Wickets  Balls
0  2008.0     Praveen Kumar,Zaheer Khan   0.0     18.0  235.2
1  2008.0  Farveez Maharoof,Amit Mishra   0.0     10.0   96.6
2  2008.0    Piyush Chawla,Irfan Pathan   0.0     10.0  120.0


# Calculate the performance metrics (e.g., runs, wickets, boundaries)


In [37]:
print(rahul_tewatia_data.columns)

Index(['season', 'id', 'name', 'short_name', 'description', 'home_team',
       'away_team', 'toss_won', 'decision', '1st_inning_score',
       '2nd_inning_score', 'winner', 'result', 'start_date', 'end_date',
       'venue_id', 'venue_name', 'home_captain', 'away_captain', 'pom',
       'points', 'super_over', 'home_overs', 'home_runs', 'home_wickets',
       'home_boundaries', 'away_overs', 'away_runs', 'away_wickets',
       'away_boundaries', 'highlights', 'home_key_batsman', 'home_key_bowler',
       'home_playx1', 'away_playx1', 'away_key_batsman', 'away_key_bowler',
       'match_days', 'umpire1', 'umpire2', 'tv_umpire', 'referee',
       'reserve_umpire'],
      dtype='object')


In [38]:
rahul_tewatia_performance = rahul_tewatia_data.groupby('season')[['home_runs', 'home_wickets', 'home_boundaries', 'away_runs', 'away_wickets', 'away_boundaries']].sum()

# Calculate the average performance metrics

In [39]:
average_performance = rahul_tewatia_performance.mean()

# Print the relationship between performance and salary

In [40]:
print("Relationship between Rahul Tewatia's performance and salary:")
print("Salary: 9 crore")
print("Average runs: ", average_performance['home_runs'] + average_performance['away_runs'])
print("Average wickets: ", average_performance['home_wickets'] + average_performance['away_wickets'])
print("Average boundaries: ", average_performance['home_boundaries'] + average_performance['away_boundaries'])

Relationship between Rahul Tewatia's performance and salary:
Salary: 9 crore
Average runs:  2878.777777777778
Average wickets:  101.11111111111111
Average boundaries:  359.1111111111111


###This output shows the average performance of Rahul Tewatia in matches where his salary is 9 crore. The average runs scored by Tewatia in these matches is approximately 2878.78, the average wickets taken is approximately 101.11, and the average boundaries hit is approximately 359.11.

This information can help in understanding the relationship between Tewatia's performance and his salary. If his performance is consistently high, it could indicate that he is being paid a fair salary. However, if his performance is low compared to his salary, it could suggest that he is overpaid. Further analysis would be needed to make a definitive conclusio
n.

In [69]:
import scipy.stats as stats
import matplotlib.pyplot as plt
import numpy as np

In [92]:
last_three_years = ipl_data_player[(ipl_data_player['season'] >= 2022) & (ipl_data_player['season'] <= 2024)]
performance_metrics = last_three_years.groupby('Player').agg({'Runs': 'sum', 'Wickets': 'sum'})

In [93]:
ipl_data_grouped = ipl_data_player.groupby(['season', 'Player']).agg({'Runs': 'sum', 'Wickets': 'sum', 'Balls': 'sum'}).reset_index()

In [95]:
df = pd.DataFrame(data)

In [94]:
data = {
    'Player': [
        'Abhishek Porel', 'Anrich Nortje', 'Axar Patel', 'David Warner', 'Ishant Sharma',
        'Kuldeep Yadav', 'Lalit Yadav', 'Lungi Ngidi', 'Mitchell Marsh', 'Mukesh Kumar',
        'Pravin Dubey', 'Prithvi Shaw', 'Rishabh Pant', 'Khaleel Ahmed', 'Vicky Ostwal',
        'Yash Dhull', 'Ajay Mandal', 'Ajinkya Rahane', 'Deepak Chahar', 'Devon Conway',
        'Maheesh Theekshana', 'Matheesha Pathirana', 'Mitchell Santner', 'Moeen Ali',
        'MS Dhoni', 'Mukesh Choudhary', 'Nishant Sindhu', 'Prashant Solanki',
        'Rajvardhan Hangargekar', 'Ravindra Jadeja', 'Ruturaj Gaikwad', 'Shaik Rasheed',
        'Shivam Dube', 'Simarjeet Singh', 'Tushar Deshpande', 'Abhinav Sadarangani',
        'B. Sai Sudharsan', 'Darshan Nalkande', 'David Miller', 'Jayant Yadav',
        'Joshua Little', 'Kane Williamson', 'Matthew Wade', 'Mohammad Shami', 'Mohit Sharma',
        'Noor Ahmad', 'R. Sai Kishore', 'Rahul Tewatia', 'Rashid Khan', 'Shubman Gill',
        'Vijay Shankar', 'Shreyas Iyer', 'Nitish Rana', 'Venkatesh Iyer', 'Andre Russell',
        'Sunil Narine', 'Harshit Rana', 'Varun Chakravarthy', 'Anukul Roy', 'Rinku Singh',
        'Rahmanullah Gurbaz', 'Amit Mishra', 'Ayush Badoni', 'Deepak Hooda', 'Devdutt Padikkal (T)',
        'K. Gowtham', 'KL Rahul', 'Krunal Pandya', 'Kyle Mayers', 'Marcus Stoinis',
        'Mark Wood', 'Mayank Yadav', 'Mohsin Khan', 'Naveen Ul Haq', 'Nicholas Pooran',
        'Prerak Mankad', 'Quinton De Kock', 'Ravi Bishnoi', 'Yash Thakur', 'Akash Madhwal',
        'Arjun Tendulkar', 'Dewald Brevis', 'Ishan Kishan', 'Hardik Pandya (T)', 'Jason Behrendorff',
        'Jasprit Bumrah', 'Kumar Kartikeya Singh', 'Tilak Varma', 'Nehal Wadhera',
        'Piyush Chawla', 'Rohit Sharma', 'Romario Shepherd (T)', 'Shams Mulani', 'Surya Kumar Yadav',
        'Tim David', 'Vishnu Vinod', 'Arshdeep Singh', 'Atharva Taide', 'Harpreet  Brar',
        'Harpreet Bhatia', 'Jitesh Sharma', 'Jonny Bairstow', 'Kagiso Rabada', 'Liam Livingstone',
        'Nathan Ellis', 'Prabhsimran Singh', 'Rahul Chahar', 'Rishi Dhawan', 'Sam Curran',
        'Shikhar Dhawan', 'Shivam  Singh', 'Sikandar Raza', 'Vidwath Kaverappa', 'Adam Zampa',
        'Avesh Khan (T)', 'Dhruv Jurel', 'Donovan Ferreira', 'Jos Buttler', 'Kuldeep  Sen',
        'Kunal Rathore', 'Navdeep Saini', 'Prasidh Krishna', 'R. Ashwin', 'Riyan Parag',
        'Sandeep Sharma', 'Sanju Samson', 'Shimron Hetmyer', 'Trent Boult', 'Yashaswi Jaiswal',
        'Yuzvendra Chahal', 'Akash Deep', 'Anuj Rawat', 'Dinesh  Karthik', 'Faf Du Plessis',
        'Glenn Maxwell', 'Himanshu Sharma', 'Karn Sharma', 'Mahipal Lomror', 'Manoj Bhandage',
        'Mayank Dagar (T)', 'Mohammed Siraj', 'Rajan Kumar', 'Rajat Patidar', 'Virat Kohli',
        'Vyshak Vijay Kumar', 'Will Jacks', 'Cameron Green (T)', 'Abdul Samad', 'Abhishek Sharma',
        'Aiden Markram', 'Anmolpreet Singh', 'Bhuvneshwar Kumar', 'Fazalhaq Farooqi',
        'Glenn Phillips', 'Heinrich Klaasen', 'Marco Jansen', 'Mayank Agarwal', 'Mayank Markande',
        'Nitish Kumar Reddy', 'Rahul Tripathi', 'Sanvir  Singh', 'Shahbaz Ahamad (T)', 'T. Natarajan',
        'Umran Malik', 'Upendra Singh Yadav', 'Washington Sundar'
    ],
    'Salary': [
        '20 lakh', '6.5 crore', '9 crore', '6.25 crore', '50 lakh', '2 crore', '65 lakh', '5 crore', '6.5 crore', '5.5 crore',
        '50 lakh', '7.5 crore', '16 crore', '5.25 crore', '20 lakh', '50 lakh', '20 lakh', '50 lakh', '14 crore', '1 crore',
        '70 lakh', '20 lakh', '1.9 crore', '8 crore', '12 crore', '20 lakh', '60 lakh', '1.2 crore', '1.5 crore', '16 crore',
        '6 crore', '20 lakh', '4 crore', '20 lakh', '20 lakh', '2.6 crore', '20 lakh', '20 lakh', '3 crore', '1.7 crore',
        '4.4 crore', '2 crore', '2.4 crore', '6.25 crore', '50 lakhs', '30 lakh', '3 crore', '9 crore', '15 crore', '7 crore',
        '1.4 crore', '12.25 crore', '8 crore', '8 crore', '12 crore', '6 crore', '20 lakh', '8 crore', '20 lakh', '55 lakh',
        '50 lakh', '50 lakh', '20 lakh', '5.75 crore', '7.75 crore', '90 lakh', '17 crore', '8.25 crore', '50 lakh', '9.2 crore',
        '7.5 crore', '20 lakh', '20 lakh', '50 lakh', '16 crore', '20 lakh', '6.75 Crore', '4 crore', '45 lakh', '20 lakh',
        '30 lakh', '3 crore', '15.25 crore', '15 crore', '50 lakh', '12 crore', '20 lakh', '1.70 crore', '20 lakh', '50 lakh',
        '16 Crore', '50 lakh', '20 lakh', '8 crore', '8.25 Crore', '20 lakh', '4 crore', '20 lakh', '3.8 crore', '40 lakh',
        '20 lakh', '6.75 crore', '9.25 crore', '11.5 crore', '75 lakh', '60 lakh', '5.25 crore', '55 lakh', '18.50 crore',
        '8.25 crore', '20 lakh', '50 lakh', '20 lakh', '1.5 crore', '10 crore', '20 lakh', '50 lakh', '10 crore', '20 lakh',
        '20 lakh', '2.6 crore', '10 crore', '5 crore', '3.8 crore', '50 lakh', '14 crore', '8.5 crore', '8 crore', '4 crore',
        '6.5 crore', '20 lakh', '3.4 crore', '5.5 crore', '7 crore', '11 crore', '20 lakh', '50 lakh', '95 lakh', '20 lakh',
        '1.8 crore', '7 crore', '70 lakh', '20 lakh', '15 crore', '20 lakh', '3.2 crore', '17.5 crore', '4 crore', '6.5 crore',
        '2.6 crore', '20 lakh', '4.2 crore', '50 lakh', '1.5 crore', '5.25 crore', '4.2 crore', '8.25 crore', '50 lakh',
        '20 lakh', '8.5 crore', '20 lakh', '2.4 crore', '3.2 crore', '4 crore', '25 lakh', '8.75 crore'
    ]
}


In [98]:
player_salary = pd.DataFrame(data)

In [99]:
player_salary['Salary'] = player_salary['Salary'].str.replace('s', '').str.strip()

In [100]:
def convert_salary(salary):
    salary = salary.replace(',', '')
    if 'lakh' in salary.lower():
        value = float(salary.replace(' lakh', '').replace(' ', ''))
        return int(value * 100000)
    elif 'crore' in salary.lower():
        value = float(salary.lower().replace('crore', '').replace(' ', ''))
        return int(value * 10000000)
    else:
        return int(float(salary))

In [101]:
player_salary['Salary'] = player_salary['Salary'].apply(convert_salary)

In [106]:
salaries = pd.DataFrame({
    'Player': [
        'Abhishek Porel', 'Anrich Nortje', 'Axar Patel', 'David Warner', 'Ishant Sharma', 'Kuldeep Yadav', 'Lalit Yadav', 'Lungi Ngidi', 'Mitchell Marsh', 
        'Mukesh Kumar', 'Pravin Dubey', 'Prithvi Shaw', 'Rishabh Pant', 'Khaleel Ahmed', 'Vicky Ostwal', 'Yash Dhull', 'Ajay Mandal', 'Ajinkya Rahane', 
        'Deepak Chahar', 'Devon Conway', 'Maheesh Theekshana', 'Matheesha Pathirana', 'Mitchell Santner', 'Moeen Ali', 'MS Dhoni', 'Mukesh Choudhary', 
        'Nishant Sindhu', 'Prashant Solanki', 'Rajvardhan Hangargekar', 'Ravindra Jadeja', 'Ruturaj Gaikwad', 'Shaik Rasheed', 'Shivam Dube', 'Simarjeet Singh', 
        'Tushar Deshpande', 'Abhinav Sadarangani', 'B. Sai Sudharsan', 'Darshan Nalkande', 'David Miller', 'Jayant Yadav', 'Joshua Little', 'Kane Williamson', 
        'Matthew Wade', 'Mohammad Shami', 'Mohit Sharma', 'Noor Ahmad', 'R. Sai Kishore', 'Rahul Tewatia', 'Rashid Khan', 'Shubman Gill', 'Vijay Shankar', 
        'Shreyas Iyer', 'Nitish Rana', 'Venkatesh Iyer', 'Andre Russell', 'Sunil Narine', 'Harshit Rana', 'Varun Chakravarthy', 'Anukul Roy', 'Rinku Singh', 
        'Rahmanullah Gurbaz', 'Amit Mishra', 'Ayush Badoni', 'Deepak Hooda', 'Devdutt Padikkal (T)', 'K. Gowtham', 'KL Rahul', 'Krunal Pandya', 'Kyle Mayers', 
        'Marcus Stoinis', 'Mark Wood', 'Mayank Yadav', 'Mohsin Khan', 'Naveen Ul Haq', 'Nicholas Pooran', 'Prerak Mankad', 'Quinton De Kock', 'Ravi Bishnoi', 
        'Yash Thakur', 'Akash Madhwal', 'Arjun Tendulkar', 'Dewald Brevis', 'Ishan Kishan', 'Hardik Pandya (T)', 'Jason Behrendorff', 'Jasprit Bumrah', 
        'Kumar Kartikeya Singh', 'Tilak Varma', 'Nehal Wadhera', 'Piyush Chawla', 'Rohit Sharma', 'Romario Shepherd (T)', 'Shams Mulani', 'Surya Kumar Yadav', 
        'Tim David', 'Vishnu Vinod', 'Arshdeep Singh', 'Atharva Taide', 'Harpreet Brar', 'Harpreet Bhatia', 'Jitesh Sharma', 'Jonny Bairstow', 'Kagiso Rabada', 
        'Liam Livingstone', 'Nathan Ellis', 'Prabhsimran Singh', 'Rahul Chahar', 'Rishi Dhawan', 'Sam Curran', 'Shikhar Dhawan', 'Shivam Singh', 'Sikandar Raza', 
        'Vidwath Kaverappa', 'Adam Zampa', 'Avesh Khan (T)', 'Dhruv Jurel', 'Donovan Ferreira', 'Jos Buttler', 'Kuldeep Sen', 'Kunal Rathore', 'Navdeep Saini', 
        'Prasidh Krishna', 'R. Ashwin', 'Riyan Parag', 'Sandeep Sharma', 'Sanju Samson', 'Shimron Hetmyer', 'Trent Boult', 'Yashaswi Jaiswal', 'Yuzvendra Chahal', 
        'Akash Deep', 'Anuj Rawat', 'Dinesh Karthik', 'Faf Du Plessis', 'Glenn Maxwell', 'Himanshu Sharma', 'Karn Sharma', 'Mahipal Lomror', 'Manoj Bhandage', 
        'Mayank Dagar (T)', 'Mohammed Siraj', 'Rajan Kumar', 'Rajat Patidar', 'Virat Kohli', 'Vyshak Vijay Kumar', 'Will Jacks', 'Cameron Green (T)', 'Abdul Samad', 
        'Abhishek Sharma', 'Aiden Markram', 'Anmolpreet Singh', 'Bhuvneshwar Kumar', 'Fazalhaq Farooqi', 'Glenn Phillips', 'Heinrich Klaasen', 'Marco Jansen', 
        'Mayank Agarwal', 'Mayank Markande', 'Nitish Kumar Reddy', 'Rahul Tripathi', 'Sanvir Singh', 'Shahbaz Ahamad (T)', 'T. Natarajan', 'Umran Malik', 
        'Upendra Singh Yadav', 'Washington Sundar'
    ], 
    'Salary': [
        '20 lakh', '6.5 crore', '9 crore', '6.25 crore', '50 lakh', '2 crore', '65 lakh', '5 crore', '6.5 crore', '5.5 crore', '50 lakh', '7.5 crore', '16 crore', 
        '5.25 crore', '20 lakh', '50 lakh', '20 lakh', '50 lakh', '14 crore', '1 crore', '70 lakh', '20 lakh', '1.9 crore', '8 crore', '12 crore', '20 lakh', 
        '60 lakh', '1.2 crore', '1.5 crore', '16 crore', '6 crore', '20 lakh', '4 crore', '20 lakh', '20 lakh', '2.6 crore', '20 lakh', '20 lakh', '3 crore', 
        '1.7 crore', '4.4 crore', '2 crore', '2.4 crore', '6.25 crore', '50 lakhs', '30 lakh', '3 crore', '9 crore', '15 crore', '7 crore', '1.4 crore', '12.25 crore', 
        '8 crore', '8 crore', '12 crore', '6 crore', '20 lakh', '8 crore', '20 lakh', '55 lakh', '50 lakh', '50 lakh', '20 lakh', '5.75 crore', '7.75 crore', 
        '90 lakh', '17 crore', '8.25 crore', '50 lakh', '9.2 crore', '7.5 crore', '20 lakh', '20 lakh', '50 lakh', '16 crore', '20 lakh', '6.75 Crore', '4 crore', 
        '45 lakh', '20 lakh', '30 lakh', '3 crore', '15.25 crore', '15 crore', '50 lakh', '12 crore', '20 lakh', '1.70 crore', '20 lakh', '50 lakh', '16 Crore', 
        '50 lakh', '20 lakh', '8 crore', '8.25 Crore', '20 lakh', '4 crore', '20 lakh', '3.8 crore', '40 lakh', '20 lakh', '6.75 crore', '9.25 crore', '11.5 crore', 
        '75 lakh', '60 lakh', '5.25 crore', '55 lakh', '18.50 crore', '8.25 crore', '20 lakh', '50 lakh', '20 lakh', '1.5 crore', '10 crore', '20 lakh', '50 lakh', 
        '10 crore', '20 lakh', '20 lakh', '2.6 crore', '10 crore', '5 crore', '3.8 crore', '50 lakh', '14 crore', '8.5 crore', '8 crore', '4 crore', '6.5 crore', 
        '20 lakh', '3.4 crore', '5.5 crore', '7 crore', '11 crore', '20 lakh', '50 lakh', '95 lakh', '20 lakh', '1.8 crore', '7 crore', '70 lakh', '20 lakh', '15 crore', 
        '20 lakh', '3.2 crore', '17.5 crore', '4 crore', '6.5 crore', '2.6 crore', '20 lakh', '4.2 crore', '50 lakh', '1.5 crore', '5.25 crore', '4.2 crore', 
        '8.25 crore', '50 lakh', '20 lakh', '8.5 crore', '20 lakh', '2.4 crore', '3.2 crore', '4 crore', '25 lakh', '8.75 crore'
    ]
})

print(salaries)

                  Player      Salary
0         Abhishek Porel     20 lakh
1          Anrich Nortje   6.5 crore
2             Axar Patel     9 crore
3           David Warner  6.25 crore
4          Ishant Sharma     50 lakh
..                   ...         ...
161   Shahbaz Ahamad (T)   2.4 crore
162         T. Natarajan   3.2 crore
163          Umran Malik     4 crore
164  Upendra Singh Yadav     25 lakh
165    Washington Sundar  8.75 crore

[166 rows x 2 columns]


In [107]:
performance_metrics = pd.concat([performance_metrics, salaries], axis=1)

In [108]:
print(performance_metrics)

                                   Runs  Wickets               Player  \
Abdul Samad,Mayank Agarwal        131.0      0.0                  NaN   
Abhishek Sharma,Romario Shepherd  157.0      0.0                  NaN   
Adam Zampa,KM Asif                  0.0     10.0                  NaN   
Adam Zampa,Ravichandran Ashwin      0.0      5.0                  NaN   
Aiden Markram,Heinrich Klaasen    166.0      0.0                  NaN   
...                                 ...      ...                  ...   
161                                 NaN      NaN   Shahbaz Ahamad (T)   
162                                 NaN      NaN         T. Natarajan   
163                                 NaN      NaN          Umran Malik   
164                                 NaN      NaN  Upendra Singh Yadav   
165                                 NaN      NaN    Washington Sundar   

                                      Salary  
Abdul Samad,Mayank Agarwal               NaN  
Abhishek Sharma,Romario Sheph

In [86]:
def neg_binom_log_likelihood(params, x):
    k, p = params
    ll = np.sum(nbinom.logpmf(x, k, p))
    return -ll


In [87]:
top_run_getters = ipl_data_grouped.groupby('season').apply(lambda x: x.nlargest(3, 'Runs')).reset_index(drop=True)
top_wicket_takers = ipl_data_grouped.groupby('season').apply(lambda x: x.nlargest(3, 'Wickets')).reset_index(drop=True)

# Fit distributions to the runs scored and wickets taken data

In [88]:
from scipy.stats import norm, poisson, nbinom
from scipy.optimize import minimize

In [89]:
distributions = ['norm', 'poisson', 'nbinom']

for dist in distributions:
    print(f"Fitting {dist} distribution to runs scored data...")
    if dist == 'norm':
        params = norm.fit(top_run_getters['Runs'])
        print(f"Normal distribution parameters: {params[0]}, {params[1]}")
    elif dist == 'poisson':
        # Poisson parameter (lambda) is the mean of the data
        params = [top_run_getters['Runs'].mean()]
        print(f"Poisson distribution parameter (lambda): {params[0]}")
    elif dist == 'nbinom':
        x = top_run_getters['Runs']
        init_params = [1, 0.5]  # initial guess for k and p
        res = minimize(neg_binom_log_likelihood, init_params, method="Nelder-Mead", args=(x,))
        k, p = res.x
        params = (k, p)
        print(f"Negative binomial distribution parameters: {params[0]}, {params[1]}")
    print()

    print(f"Fitting {dist} distribution to wickets taken data...")
    if dist == 'norm':
        params = norm.fit(top_wicket_takers['Wickets'])
        print(f"Normal distribution parameters: {params[0]}, {params[1]}")
    elif dist == 'poisson':
        # Poisson parameter (lambda) is the mean of the data
        params = [top_wicket_takers['Wickets'].mean()]
        print(f"Poisson distribution parameter (lambda): {params[0]}")
    elif dist == 'nbinom':
        x = top_wicket_takers['Wickets']
        init_params = [1, 0.5]  # initial guess for k and p
        res = minimize(neg_binom_log_likelihood, init_params, method="Nelder-Mead", args=(x,))
        k, p = res.x
        params = (k, p)
        print(f"Negative binomial distribution parameters: {params[0]}, {params[1]}")
    print()

Fitting norm distribution to runs scored data...
Normal distribution parameters: 374.625, 87.09468626156247

Fitting norm distribution to wickets taken data...
Normal distribution parameters: 13.333333333333334, 3.009245014211298

Fitting poisson distribution to runs scored data...
Poisson distribution parameter (lambda): 374.625

Fitting poisson distribution to wickets taken data...
Poisson distribution parameter (lambda): 13.333333333333334

Fitting nbinom distribution to runs scored data...
Negative binomial distribution parameters: 21.287372978675116, 0.053767876175308676

Fitting nbinom distribution to wickets taken data...
Negative binomial distribution parameters: 957.602609938167, 0.9862885906054499



# Significant Difference Between the Salaries of the Top 10 Batsmen and Top Wicket-Taking Bowlers Over the Last Three Years

In [124]:
from scipy.stats import ttest_ind

In [125]:
salaries = pd.DataFrame({
    'Player': [
        'Abhishek Porel', 'Anrich Nortje', 'Axar Patel', 'David Warner', 'Ishant Sharma', 'Kuldeep Yadav', 'Lalit Yadav', 'Lungi Ngidi', 'Mitchell Marsh', 
        'Mukesh Kumar', 'Pravin Dubey', 'Prithvi Shaw', 'Rishabh Pant', 'Khaleel Ahmed', 'Vicky Ostwal', 'Yash Dhull', 'Ajay Mandal', 'Ajinkya Rahane', 
        'Deepak Chahar', 'Devon Conway', 'Maheesh Theekshana', 'Matheesha Pathirana', 'Mitchell Santner', 'Moeen Ali', 'MS Dhoni', 'Mukesh Choudhary', 
        'Nishant Sindhu', 'Prashant Solanki', 'Rajvardhan Hangargekar', 'Ravindra Jadeja', 'Ruturaj Gaikwad', 'Shaik Rasheed', 'Shivam Dube', 'Simarjeet Singh', 
        'Tushar Deshpande', 'Abhinav Sadarangani', 'B. Sai Sudharsan', 'Darshan Nalkande', 'David Miller', 'Jayant Yadav', 'Joshua Little', 'Kane Williamson', 
        'Matthew Wade', 'Mohammad Shami', 'Mohit Sharma', 'Noor Ahmad', 'R. Sai Kishore', 'Rahul Tewatia', 'Rashid Khan', 'Shubman Gill', 'Vijay Shankar', 
        'Shreyas Iyer', 'Nitish Rana', 'Venkatesh Iyer', 'Andre Russell', 'Sunil Narine', 'Harshit Rana', 'Varun Chakravarthy', 'Anukul Roy', 'Rinku Singh', 
        'Rahmanullah Gurbaz', 'Amit Mishra', 'Ayush Badoni', 'Deepak Hooda', 'Devdutt Padikkal (T)', 'K. Gowtham', 'KL Rahul', 'Krunal Pandya', 'Kyle Mayers', 
        'Marcus Stoinis', 'Mark Wood', 'Mayank Yadav', 'Mohsin Khan', 'Naveen Ul Haq', 'Nicholas Pooran', 'Prerak Mankad', 'Quinton De Kock', 'Ravi Bishnoi', 
        'Yash Thakur', 'Akash Madhwal', 'Arjun Tendulkar', 'Dewald Brevis', 'Ishan Kishan', 'Hardik Pandya (T)', 'Jason Behrendorff', 'Jasprit Bumrah', 
        'Kumar Kartikeya Singh', 'Tilak Varma', 'Nehal Wadhera', 'Piyush Chawla', 'Rohit Sharma', 'Romario Shepherd (T)', 'Shams Mulani', 'Surya Kumar Yadav', 
        'Tim David', 'Vishnu Vinod', 'Arshdeep Singh', 'Atharva Taide', 'Harpreet Brar', 'Harpreet Bhatia', 'Jitesh Sharma', 'Jonny Bairstow', 'Kagiso Rabada', 
        'Liam Livingstone', 'Nathan Ellis', 'Prabhsimran Singh', 'Rahul Chahar', 'Rishi Dhawan', 'Sam Curran', 'Shikhar Dhawan', 'Shivam Singh', 'Sikandar Raza', 
        'Vidwath Kaverappa', 'Adam Zampa', 'Avesh Khan (T)', 'Dhruv Jurel', 'Donovan Ferreira', 'Jos Buttler', 'Kuldeep Sen', 'Kunal Rathore', 'Navdeep Saini', 
        'Prasidh Krishna', 'R. Ashwin', 'Riyan Parag', 'Sandeep Sharma', 'Sanju Samson', 'Shimron Hetmyer', 'Trent Boult', 'Yashaswi Jaiswal', 'Yuzvendra Chahal', 
        'Akash Deep', 'Anuj Rawat', 'Dinesh Karthik', 'Faf Du Plessis', 'Glenn Maxwell', 'Himanshu Sharma', 'Karn Sharma', 'Mahipal Lomror', 'Manoj Bhandage', 
        'Mayank Dagar (T)', 'Mohammed Siraj', 'Rajan Kumar', 'Rajat Patidar', 'Virat Kohli', 'Vyshak Vijay Kumar', 'Will Jacks', 'Cameron Green (T)', 'Abdul Samad', 
        'Abhishek Sharma', 'Aiden Markram', 'Anmolpreet Singh', 'Bhuvneshwar Kumar', 'Fazalhaq Farooqi', 'Glenn Phillips', 'Heinrich Klaasen', 'Marco Jansen', 
        'Mayank Agarwal', 'Mayank Markande', 'Nitish Kumar Reddy', 'Rahul Tripathi', 'Sanvir Singh', 'Shahbaz Ahamad (T)', 'T. Natarajan', 'Umran Malik', 
        'Upendra Singh Yadav', 'Washington Sundar'
    ], 
    'Salary': [
        '20 lakh', '6.5 crore', '9 crore', '6.25 crore', '50 lakh', '2 crore', '65 lakh', '5 crore', '6.5 crore', '5.5 crore', '50 lakh', '7.5 crore', '16 crore', 
        '5.25 crore', '20 lakh', '50 lakh', '20 lakh', '50 lakh', '14 crore', '1 crore', '70 lakh', '20 lakh', '1.9 crore', '8 crore', '12 crore', '20 lakh', 
        '60 lakh', '1.2 crore', '1.5 crore', '16 crore', '6 crore', '20 lakh', '4 crore', '20 lakh', '20 lakh', '2.6 crore', '20 lakh', '20 lakh', '3 crore', 
        '1.7 crore', '4.4 crore', '2 crore', '2.4 crore', '6.25 crore', '50 lakhs', '30 lakh', '3 crore', '9 crore', '15 crore', '7 crore', '1.4 crore', '12.25 crore', 
        '8 crore', '8 crore', '12 crore', '6 crore', '20 lakh', '8 crore', '20 lakh', '55 lakh', '50 lakh', '50 lakh', '20 lakh', '5.75 crore', '7.75 crore', 
        '90 lakh', '17 crore', '8.25 crore', '50 lakh', '9.2 crore', '7.5 crore', '20 lakh', '20 lakh', '50 lakh', '16 crore', '20 lakh', '6.75 Crore', '4 crore', 
        '45 lakh', '20 lakh', '30 lakh', '3 crore', '15.25 crore', '15 crore', '50 lakh', '12 crore', '20 lakh', '1.70 crore', '20 lakh', '50 lakh', '16 Crore', 
        '50 lakh', '20 lakh', '8 crore', '8.25 Crore', '20 lakh', '4 crore', '20 lakh', '3.8 crore', '40 lakh', '20 lakh', '6.75 crore', '9.25 crore', '11.5 crore', 
        '75 lakh', '60 lakh', '5.25 crore', '55 lakh', '18.50 crore', '8.25 crore', '20 lakh', '50 lakh', '20 lakh', '1.5 crore', '10 crore', '20 lakh', '50 lakh', 
        '10 crore', '20 lakh', '20 lakh', '2.6 crore', '10 crore', '5 crore', '3.8 crore', '50 lakh', '14 crore', '8.5 crore', '8 crore', '4 crore', '6.5 crore', 
        '20 lakh', '3.4 crore', '5.5 crore', '7 crore', '11 crore', '20 lakh', '50 lakh', '95 lakh', '20 lakh', '1.8 crore', '7 crore', '70 lakh', '20 lakh', '15 crore', 
        '20 lakh', '3.2 crore', '17.5 crore', '4 crore', '6.5 crore', '2.6 crore', '20 lakh', '4.2 crore', '50 lakh', '1.5 crore', '5.25 crore', '4.2 crore', 
        '8.25 crore', '50 lakh', '20 lakh', '8.5 crore', '20 lakh', '2.4 crore', '3.2 crore', '4 crore', '25 lakh', '8.75 crore'
    ]
})

In [129]:
player_salary = pd.DataFrame(data)

In [130]:
player_salary['Salary'] = player_salary['Salary'].str.replace('s', '').str.strip()

In [131]:
def convert_salary(salary):
    salary = salary.replace(',', '')
    if 'lakh' in salary.lower():
        value = float(salary.replace(' lakh', '').replace(' ', ''))
        return int(value * 100000)
    elif 'crore' in salary.lower():
        value = float(salary.lower().replace('crore', '').replace(' ', ''))
        return int(value * 10000000)
    else:
        return int(float(salary))

In [132]:
player_salary['Salary'] = player_salary['Salary'].apply(convert_salary)

In [142]:
top_10_batsmen = performance_metrics.nlargest(10, 'Runs')
top_10_bowlers = performance_metrics.nlargest(10, 'Wickets')

In [143]:
batsmen = ['David Warner', 'Prithvi Shaw', 'Rishabh Pant', 'Ajinkya Rahane', 'Devon Conway', 
           'Mitchell Marsh', 'Venkatesh Iyer', 'Nitish Rana', 'Shubman Gill', 'Vijay Shankar',
           'Shreyas Iyer', 'Rahmanullah Gurbaz', 'Nicholas Pooran', 'Quinton De Kock', 
           'Virat Kohli', 'Faf Du Plessis', 'Glenn Maxwell', 'Shikhar Dhawan', 'Mayank Agarwal',
           'Jos Buttler', 'Sanju Samson', 'Rohit Sharma', 'Suryakumar Yadav']

bowlers = ['Anrich Nortje', 'Ishant Sharma', 'Kuldeep Yadav', 'Lungi Ngidi', 'Khaleel Ahmed',
           'Mukesh Kumar', 'Mohammad Shami', 'Mohit Sharma', 'Rashid Khan', 'Jayant Yadav', 
           'Ravi Bishnoi', 'Jasprit Bumrah', 'Piyush Chawla', 'Arshdeep Singh', 'Kagiso Rabada',
           'Rahul Chahar', 'Umran Malik', 'Yuzvendra Chahal', 'Harshal Patel', 'Mohammed Siraj']


In [144]:
t_stat, p_value = ttest_ind(salaries_batsmen['Salary'], salaries_bowlers['Salary'])
print(f"t-statistic: {t_stat}")
print(f"p-value: {p_value}")


t-statistic: nan
p-value: nan


In [145]:
if p_value < 0.05:
    print("There is a significant difference in salaries between the top 10 batsmen and top 10 bowlers.")
else:
    print("There is no significant difference in salaries between the top 10 batsmen and top 10 bowlers.")

There is no significant difference in salaries between the top 10 batsmen and top 10 bowlers.
